internal/storage/pod.go

```diff
@@ -2,47 +2,206 @@ package storage
+		}
+
+		podOutput, _, err := execPod(storagePod.Name, namespace.Name, cmd, stdin, false)
+		if err != nil {
+			opLog.Info(fmt.Sprintf("error checking storage-calculator pod %s/%s for volume %s size: %v", namespace.Name, storagePod.Name, vol.Name, err))
+			continue
+		}
+
+		kiBytes := strings.TrimSpace(podOutput)
+		kiBytesInt, _ := strconv.Atoi(kiBytes)
+		storageClaims = append(storageClaims, StorageClaim{
+			Environment:          environmentID,
+			PersisteStorageClaim: vol.Name,
+			KiBUsed:              uint64(kiBytesInt),
+		})
+	}
+
+	c.cleanup(ctx, opLog, storagePod)
+
+	return storageClaims, nil
+}
+
+// createDatabasePod creates a pod that will be used to calculate storage usage for databases listed
+// in `services`.
+func (c *Calculator) createDatabasePod(
 	ctx context.Context,
 	opLog logr.Logger,
 	namespace corev1.Namespace,
-	spn storageCalculatorPod,
+	services databasesCalculatorPod,
 	environmentID int,
-	ignoreRegex string,
-	checkedDatabase *bool,
-) error {
-	storData := ActionData{}
-	// define the storage-calculator pod
-	podName := fmt.Sprintf("storage-calculator-%s", ns.RandString(8))
+) ([]StorageClaim, error) {
+	storageClaims := []StorageClaim{}
+	var stdin io.Reader
+	storagePod := c.getPodSpec(ctx, opLog, namespace, "")
+
+	opLog.Info(fmt.Sprintf("creating storage-calculator pod %s/%s", namespace.Name, storagePod.Name))
+	if err := c.Client.Create(ctx, storagePod); err != nil {
+		return []StorageClaim{}, fmt.Errorf("error creating storage-calculator pod %s/%s: %v", namespace.Name, storagePod.Name, err)
+	}
+
+	// Wait some time for the pod to start.
+	if err := wait.PollUntilContextTimeout(ctx, time.Second, 90*time.Second, true,
+		c.hasRunningPod(ctx, namespace.Name, storagePod.Name)); err != nil {
+		c.cleanup(ctx, opLog, storagePod)
+		return []StorageClaim{}, fmt.Errorf("error starting storage-calculator pod %s/%s: %v", namespace.Name, storagePod.Name, err)
+	}
+
+	// Calculate the database sizes.
+	for _, mariadb := range services.MariaDB {
+		cmd := []string{
+			"/bin/sh",
+			"-c",
+			fmt.Sprintf(
+				`mysql -N -s -h "%s" -u"%s" -p"%s" -P%s -e 'SELECT ROUND(SUM(data_length + index_length) / 1024, 0) FROM information_schema.tables'`,
+				mariadb.Spec.Provider.Hostname,
+				mariadb.Spec.Consumer.Username,
+				mariadb.Spec.Consumer.Password,
+				mariadb.Spec.Provider.Port,
+			)}
+
+		podOutput, _, err := execPod(storagePod.Name, namespace.Name, cmd, stdin, false)
+		if err != nil {
+			opLog.Info(fmt.Sprintf("error checking storage-calculator pod %s/%s for %s size: %v", namespace.Name, storagePod.Name, mariadb.Name, err))
+			continue
+		}
+
+		if podOutput != "" {
+			kiBytes := strings.TrimSpace(podOutput)
+			kiBytesInt, _ := strconv.Atoi(kiBytes)
+			storageClaims = append(storageClaims, StorageClaim{
+				Environment: environmentID,
+				// Prevent conflicts with database pod volume names.
+				PersisteStorageClaim: fmt.Sprintf("managed-%s", mariadb.Name),
```

Comment:
I wonder if `dbaas` prefix would be better than `managed`. I get the idea behind managed, but dbaas makes it clear.

Reply:
Another reason for `dbaas` is that `managed` isn't used anywhere when referencing these service types in Lagoon generally from a user interacting with them in configuration. We use `x-dbaas` as the service types for instance. If it was `x-managed` as the service type, then I'd be more on board for managed being used.

Reply:
I simply disagree. The explanations we've had to provide multiple times and the extra documentation we've had to produce says to me that `dbaas` is not obvious.

But it's subjective, and I can workaround the problem I was trying to avoid in a non-subjective way as a future followup.